dtls/client-dtls-nonblocking.c

```diff
@@ -202,61 +110,106 @@ int main (int argc, char** argv)
     wolfSSL_set_fd(ssl, sockfd);
     wolfSSL_set_using_nonblock(ssl, 1);
     fcntl(sockfd, F_SETFL, O_NONBLOCK);
-    NonBlockingDTLS_Connect(ssl);
 
-    DatagramClient(ssl);
-    while ( (wolfSSL_write(ssl, srTest, sizeof(srTest))) != sizeof(srTest)) {
-	    printf("failed to write");
-        return 1;
-    }
+/*****************************************************************************/
+/*                     Non-blocking code for DTLS connect                    */
 
-    session = wolfSSL_get_session(ssl);
-    sslResume = wolfSSL_new(ctx);
+    while (ret != SSL_SUCCESS && (error == SSL_ERROR_WANT_READ ||
+                error == SSL_ERROR_WANT_WRITE)) {
+        
+        /* Variables that will reset upon every iteration */
+        int             currTimeout = 1;
+        int             nfds = nb_sockfd +1;
+        int             result;
+        fd_set          recvfds, errfds;
+        struct timeval  timeout = { (currTimeout > 0) ? currTimeout : 0, 0};
+        
+	    if (error == SSL_ERROR_WANT_READ) {
+	        printf("... client would read block\n");
+        }
+	    else {
+            printf("... client would write block\n");
+        }
 
-    wolfSSL_shutdown(ssl);
-    wolfSSL_free(ssl);
-    close(sockfd);
+	    currTimeout = wolfSSL_dtls_get_current_timeout(ssl);
 
-    memset(&servAddr, 0, sizeof(servAddr));
-    servAddr.sin_family = AF_INET;
-    servAddr.sin_port = htons(SERV_PORT);
-    if (inet_pton(AF_INET, host, &servAddr.sin_addr) < 1) {
-        printf("Error and/or invalid IP address");
-        return 1;
-    }
+        /* Tcp select using dtls nonblocking functionality */
+        FD_ZERO(&recvfds);
+        FD_SET(nb_sockfd, &recvfds);
+        FD_ZERO(&errfds);
+        FD_SET(nb_sockfd, &errfds);
 
-    wolfSSL_dtls_set_peer(sslResume, &servAddr, sizeof(servAddr));
+        result = select(nfds, &recvfds, NULL, &errfds, &timeout);
 
-    if ( (sockfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
-	    printf("cannot create a socket.");
-        return 1;
-    }
+        select_ret = TEST_SELECT_FAIL;
 
-    wolfSSL_set_fd(sslResume, sockfd);
-    wolfSSL_set_session(sslResume, session);
-    wolfSSL_set_using_nonblock(sslResume, 1);
-    fcntl(sockfd, F_SETFL, O_NONBLOCK);
-    NonBlockingDTLS_Connect(sslResume);
+        if (result == 0) {
+	        select_ret = TEST_TIMEOUT;
```

Comment:
Spacing is still off here.  This line is indented 7 spaces past the if.  Should be 4.